names.lua

```diff
@@ -1,68 +1,118 @@
---rename items or units
+--rename items or units with the native interface
 --[====[
 
 names
 =====
+
 Rename units or items.  Usage:
+:-help:    print this help message
+:-if a first name is desired press f, leave blank to clear current first name
+:-if viewing an artifact you can rename it
+:-if viewing a unit you can rename them
 
-:-help:     print this help message
-:-item:     if viewing an item
-:-unit:     if viewing a unit
-:-first [Somename | "Some Names like This":
-            if a first name is desired, leave blank to clear current first name
 ]====]
+
+local gui = require 'gui'
+local dlg = require 'gui.dialogs'
+local widgets = require 'gui.widgets'
 local utils = require 'utils'
+local vw
```

Comment:
You don't really have to do this. Global variables in scripts won't leak outside of scripts. Also, these should probably be members of `namescr` instead of globals anyway, and you can replace `vw` with `self._native.parent` in namescr functions (I think), which is the only place you're using `vw` anyway.

Reply:
I could swear I did this before moving the vw definition into the namescr setup actually.